Assets/Scripts/Gameplay/MushroomTrigger.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MushroomTrigger : MonoBehaviour
+{
+    private void OnTriggerStay(Collider other)
+    {
+        if(other.CompareTag("Player"))
+        {
+            DefuseBomb test = other.GetComponent<DefuseBomb>();
+            if (test != null)
+            {
+                test.AllowBombDefusal();
+            }
+        }
```

Comment:
`AllowBombDefusal` seems to never be set to false after exiting the trigger. Instead of having this script at all, why not just set the boolean directly in your `DefuseBomb` script, when the button is pressed? In other words check if the player is near enough in `DefuseBomb.HandleDefuseBomb` instead.